modules/javafx.base/src/test/java/test/javafx/util/converter/LocalDateTimeStringConverterTest.java

```diff
@@ -53,44 +58,102 @@ public class LocalDateTimeStringConverterTest {
+    private LocalDateTime validDateTime;
+
+    private LocalDateTimeStringConverter converter;
+    private Locale locale;
     private DateTimeFormatter formatter, parser;
 
-    private LocalDateTime validDateTime;
 
-    public LocalDateTimeStringConverterTest(LocalDateTimeStringConverter converter, Locale locale, FormatStyle dateStyle, FormatStyle timeStyle, LocalDateTime validDateTime, DateTimeFormatter formatter, DateTimeFormatter parser) {
-        this.converter = converter;
-        this.locale = locale;
+    public LocalDateTimeStringConverterTest(LocalDateTimeStringConverterVariant converterVariant, FormatStyle dateStyle, FormatStyle timeStyle, LocalDateTime validDateTime) {
+        this.converterVariant = converterVariant;
         this.dateStyle = dateStyle;
         this.timeStyle = timeStyle;
         this.validDateTime = validDateTime;
-        this.formatter = formatter;
-        this.parser = parser;
+
+        this.converter = null;
+        this.locale = null;
+        this.formatter = null;
+        this.parser = null;
+    }
+
+    @BeforeClass
+    public static void setupBeforeAll() {
+        // Tests require that default locale is en_US
+        oldLocale = Locale.getDefault();
+        Locale.setDefault(Locale.US);
+
+        // DateTimeFormatter uses default locale, so we can init this after updating locale
+        aFormatter = DateTimeFormatter.ofPattern("dd MM yyyy HH mm ss");
+        aParser = DateTimeFormatter.ofPattern("yyyy MM dd hh mm ss a");
+    }
+
+    @AfterClass
+    public static void teardownAfterAll() {
+        // Restore VM's old locale
+        Locale.setDefault(oldLocale);
+    }
+
+    @Before
+    public void setup() {
+        if (this.converter == null) {
+            // Locale is established now, so we can allocate objects depending on it
+            switch (this.converterVariant) {
+            case NO_PARAM:
+                this.converter = new LocalDateTimeStringConverter();
+                this.locale = Locale.getDefault(Locale.Category.FORMAT);
+                // this.formatter and this.parser remain null
+                break;
+            case WITH_FORMATTER_PARSER:
+                this.converter = new LocalDateTimeStringConverter(aFormatter, aParser);
+                this.locale = Locale.getDefault(Locale.Category.FORMAT);
+                this.formatter = aFormatter;
+                this.parser = aParser;
+                break;
+            case WITH_FORMAT_STYLES:
+                this.converter = new LocalDateTimeStringConverter(FormatStyle.SHORT, FormatStyle.SHORT, Locale.UK, IsoChronology.INSTANCE);
+                this.locale = Locale.UK;
+                // this.formatter and this.parser remain null
+                break;
+            default:
+                throw new InvalidParameterException("Invalid converter variant: " + this.converterVariant.toString());
+            }
+        }
+    }
+
+    @After
+    public void teardown() {
```

Comment:
is this method empty?  can it be removed?

Reply:
Done